inc/audiences/namespace.php

```diff
@@ -337,38 +295,34 @@ function register_field( string $field, string $label ) {
  * Queue up the audience admin UI scripts.
  */
 function admin_enqueue_scripts() {
-	if ( get_current_screen()->post_type !== POST_TYPE ) {
-		return;
-	}
 
-	wp_dequeue_script( 'post' );
-
-	wp_enqueue_script(
+	wp_register_script(
 		'altis-analytics-audience-ui',
 		Utils\get_asset_url( 'audiences.js' ),
 		[
 			'lodash',
 			'react',
 			'react-dom',
+			'wp-core-data',
 			'wp-i18n',
 			'wp-hooks',
 			'wp-data',
 			'wp-components',
 			'wp-api-fetch',
+			'wp-url',
+			'wp-compose',
 		],
 		null,
 		true
 	);
 
-	wp_enqueue_style( 'wp-components' );
-
 	// Hydrate with data to speed up the front end.
 	$data = [
 		'Fields' => get_field_data(),
 	];
 
 	// Add post data server side to load front end quickly on legacy edit screens.
-	if ( isset( $_GET['post'] ) ) {
+	if ( isset( $_GET['edit'] ) && get_post_type( intval( $_GET['edit'] ) ) === POST_TYPE && current_user_can( 'edit_audience', intval( $_GET['edit'] ) ) ) {
 		$response = rest_do_request( sprintf( '/wp/v2/audiences/%d', $_GET['post'] ) );
```

Comment:
You're `intval`ing the checks above, but not the param here; sprintf should handle it, but just in case (because I've been bitten before), suggest adding an extra `intval` to this param.

Reply:
```suggestion
		$response = rest_do_request( sprintf( '/wp/v2/audiences/%d', intval( $_GET['post'] ) ) );
```